### DISCUSSION TITLE
How to be compatible with OSC 133?

### DISCUSSION THREAD

**Discussion by ymotongpoo** [starship/starship#4221] [Q&A]
Modern terminal emulators are supporting [OSC 133](https://gitlab.freedesktop.org/Per_Bothner/specifications/blob/master/proposals/semantic-prompts.md) to express semantic markings of commands on shells and REPLs.

* [iTerm2](https://iterm2.com/documentation-escape-codes.html)
* [WezTerm](https://github.com/wez/wezterm/issues/115)
* [Alacritty](https://github.com/alacritty/alacritty/issues/5850)
* [VS Code](https://code.visualstudio.com/docs/terminal/shell-integration)

Because I use multiple OS and shells (bash and zsh), I'm happily using Starship as common prompt line generator, but I couldn't find the way to set up OSC 133. I have looked up [the document](https://starship.rs/config/), [issues](https://github.com/starship/starship/issues?q=is%3Aissue+osc+133) and [discussions](https://github.com/starship/starship/discussions?discussions_q=OSC+133) but it seems no information on how to set up OSC 133 escape sequences using Starship.

Does anyone in this community know the way to set up it?

**Comment by ymotongpoo** [comment]
Additional context: I'm currently adding these lines to zshrc to generate OSC 133 in prompts but what I'm looking at is to achieve this only with Starship.
https://gitlab.freedesktop.org/Per_Bothner/specifications/-/blob/master/proposals/prompts-data/shell-integration.zsh

**Comment by davidkna** [comment]
There hasn't been any request for such a feature yet. In zsh you can likely just run this after starship has initialized:
```sh
PS1=$'%{\e]133;P;k=i\a%}'$PS1$'%{\e]133;B\a\e]122;> \a%}'
PS2=$'%{\e]133;P;k=s\a%}'$PS2$'%{\e]133;B\a%}'
```

You can also try adding these escapes to `format` and `right_format` in the config, but this might influence internal width calculations.

**Reply by ymotongpoo** [reply]
Thank you @davidkna for the comments. Yes, as I mentioned in the previous comment, I set up the script in zshrc, but I thought it would be great if Starship can support this because I use it for multiple OS and multiple shells.
